loader/loader.go

```diff
@@ -486,3 +508,18 @@ func (l *loader) GetMatchingComponentSnapshotsForPRGroupHash(ctx context.Context
 	}
 	return &snapshots.Items, nil
 }
+
+// function GetAllIntegrationPipelineRunsForSnapshot returns list of integration pipelineruns that matches the required labels specified in the function
+func (l *loader) GetAllIntegrationPipelineRunsForSnapshot(ctx context.Context, adapterClient client.Client, snapshot *applicationapiv1alpha1.Snapshot) ([]tektonv1.PipelineRun, error) {
+	integrationPipelineRuns := &tektonv1.PipelineRunList{}
+	opts := []client.ListOption{
+		client.InNamespace(snapshot.Namespace),
+		client.MatchingLabels{
+			"pipelines.appstudio.openshift.io/type": "test",
+			"appstudio.openshift.io/snapshot":       snapshot.Name,
+		},
+	}
+	err := adapterClient.List(ctx, integrationPipelineRuns, opts...)
```

Comment:
Should we check `if err != nil` before `return integrationPipelineRuns.Items, err`?